### PR TITLE
Added findHighestInWindow in MSSpectrum. Changed PrecursorCorrection …

### PR DESCRIPTION
**Pull request by JeeH-K**
…accordingly.

A piece of PrecursorCorrection code is extracted to findHighestInWindow for further use.

### CODE REVIEW

File: src/openms/source/FILTERING/CALIBRATION/PrecursorCorrection.cpp [hunk 1]
```diff
@@ -201,23 +201,34 @@ namespace OpenMS
...
+//        MSSpectrum::ConstIterator max_intensity_it = std::max_element(left, right, Peak1D::IntensityLess());
+//
+//        // find peak (index) with highest intensity to expected position
+//        Size highest_peak_idx = max_intensity_it - rt_it->begin();
+
+        // get tolerance window and index of highest peak
         std::pair<double,double> tolerance_window = Math::getTolWindow(mz, mz_tolerance, ppm);
-        MSSpectrum::ConstIterator left = rt_it->MZBegin(tolerance_window.first);
-        MSSpectrum::ConstIterator right = rt_it->MZEnd(tolerance_window.second);
+        auto highest_peak_idx = rt_it->findHighestInWindow(mz, mz-tolerance_window.first, mz+tolerance_window.second);
```
Comment (timosachsenberg) [inline src/openms/source/FILTERING/CALIBRATION/PrecursorCorrection.cpp]: I would use Int here instead of auto. It's even shorter ;)

Comment (timosachsenberg) [reply src/openms/source/FILTERING/CALIBRATION/PrecursorCorrection.cpp]: if tests pass you can also remove the code in comments

**Comment by timosachsenberg** [comment]
looks good!
two things are missing:

- [ ] pyOpenMS bindings to the new function (see the MSSpectrum.pxd file)
- [ ] a test

File: src/openms/source/FILTERING/CALIBRATION/PrecursorCorrection.cpp (hunk 1 above)
Comment (JeeH-K) [reply src/openms/source/FILTERING/CALIBRATION/PrecursorCorrection.cpp]: sure, thanks for the suggestion

**Pull request merged**